### ISSUE TITLE
There is no role named `admin`.  

### ISSUE THREAD

**Issue by a1051575078** [spatie/laravel-permission#1544]
I need to create multiple guards. I follow the documentation prompts. Still reporting an error. Thank you
i have admin、user，guard。but  my 'admin' gurad can assign roles to users     'user('api')' gurad cant assign roles to users. roles and permissions can be created normally.
this my code 
auth.php
![01](https://user-images.githubusercontent.com/43447716/89882809-ec899280-dbf9-11ea-85b1-e6e22f374057.png)
user.php model
![02](https://user-images.githubusercontent.com/43447716/89882922-15aa2300-dbfa-11ea-87d3-215fb9efbfb9.png)
admin.php model
![03](https://user-images.githubusercontent.com/43447716/89882962-235fa880-dbfa-11ea-81cc-9f0d0d4aa2f2.png)
Run code
![04](https://user-images.githubusercontent.com/43447716/89883274-818c8b80-dbfa-11ea-8c55-0efab38e7bb3.png)
  thank u tell me my false .

**Comment by a1051575078** [comment]
and. if my user.php>$guard_name  is: 'admin'.   can use .    but .  only same   admin.php created role name .    
such us :my admin role name ='admin ' and i user.php>$guard_name='admin' and  my run code assignRole('admin') can created... 
if my admin role name  not have 'admin1'   but my run code created role name is:'admin1' and  $user->assignRole('admin1') cant assign. still error  There is no role named `admin1`. , exception: "Spatie\Permission\Exceptions\RoleDoesNotExist". databases have role name 'admin1' .

**Comment by a1051575078** [comment]
I used the tenancy/hyn saas package. This is my database structure
![01](https://user-images.githubusercontent.com/43447716/89884829-a84bc180-dbfc-11ea-9b4c-237c10826d3e.png)

**Comment by drbyte** [comment]
In the code in your screenshot, you show the following snippet, where you create the 'admin' role and save it into the `$role` variable. Later you `assignRole('admin')` which forces a string lookup, but you could `assignRole($role)` to  allow it to assign to the actual $role object you created (which avoids the string lookup).
```diff
$role=Role::create(['name' => 'admin']);
// .. givePermissionTo
foreach($request->admin as $admin){
    $val=User::create([ ....]);
    $val->givePermissionTo('.....');
-    $val->assignRole('admin');
+    $val->assignRole($role);
}
```

**Comment by a1051575078** [comment]
thank you. But I found the problem. It is because I modified the /config/permission.php >models >'permission/role' file. Forgot to use php artisan config:clear. Because I installed the tenancy/hyn dependency package. Need to switch the created tenant. Until I dd()  the Models\Role.php>create method. I found that static:: is the role of my system. Not the tenant’s.then I remembered using the php artisan config:clear command. excuse me. Thank you for your answer
![01](https://user-images.githubusercontent.com/43447716/89975448-5a849700-dc98-11ea-9254-049e2bcd3824.png)
![02](https://user-images.githubusercontent.com/43447716/89975452-5c4e5a80-dc98-11ea-9b23-9e6faff280bf.png)
![03](https://user-images.githubusercontent.com/43447716/89975456-5c4e5a80-dc98-11ea-9cf1-a03ce9f9fbcb.png)
